[@vanilla-extract/next-plugin] css-loader `url` option not respected

### Describe the bug

Using the newer `@vanilla-extract/next-plugin` versions 2.2.0 or 2.2.1, it appears that the css-loader loader used by vanilla extract:

https://github.com/vanilla-extract-css/vanilla-extract/blob/40d7e72c041989a4dc847dc66b94a4c680b5536e/packages/next-plugin/src/index.ts#L58

no longer respects/ uses the [url](https://webpack.js.org/loaders/css-loader/#url) option setup by next, which is defined here: 

- usage: https://github.com/vercel/next.js/blob/a4f2bbbe2047d4ed88e9b6f32f6b0adfc8d0c46a/packages/next/src/build/webpack/config/blocks/css/loaders/global.ts#L35-L36
- definition: https://github.com/vercel/next.js/blob/a4f2bbbe2047d4ed88e9b6f32f6b0adfc8d0c46a/packages/next/src/build/webpack/config/blocks/css/loaders/file-resolve.ts#L1-L13

This leads to the following css failing to build with a failed to resolve module error:

```ts
export const test = style({
    background: 'url("/foo/bar.png")',
});
```

likewise for anything that uses url, such as fontFace:

```ts
import { fontFace } from '@vanilla-extract/css';

const comicSans = fontFace({
  src: 'url("Comic Sans MS")'
});
```






### Reproduction

https://codesandbox.io/p/sandbox/withered-bush-5yt993

### System Info

```shell
reproduceable on code sandbox
```


### Used Package Manager

npm

### Logs

```shell
./node_modules/@vanilla-extract/webpack-plugin/extracted.js:4:0
Module not found: Can't resolve '/workspace/hello/world.png'
```


### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/.changeset/calm-starfishes-smell.md
+++ b/.changeset/calm-starfishes-smell.md
@@ -2,4 +2,4 @@
 '@vanilla-extract/next-plugin': patch
 ---
 
-fixes #1154, url statements in vanilla css would not respect next's setup for resolving assets
+Fix URL statements not not respecting Next's setup for resolving assets

--- a/.changeset/calm-starfishes-smell.md
+++ b/.changeset/calm-starfishes-smell.md
@@ -1,0 +1,5 @@
+---
+'@vanilla-extract/next-plugin': patch
+---
+
+fixes #1154, url statements in vanilla css would not respect next's setup for resolving assets

--- a/packages/next-plugin/src/index.ts
+++ b/packages/next-plugin/src/index.ts
@@ -2,6 +2,7 @@ import { VanillaExtractPlugin } from '@vanilla-extract/webpack-plugin';
 import browserslist from 'browserslist';
 import { lazyPostCSS } from 'next/dist/build/webpack/config/blocks/css';
 import { findPagesDir } from 'next/dist/lib/find-pages-dir';
+import { cssFileResolve } from 'next/dist/build/webpack/config/blocks/css/loaders/file-resolve';
 import NextMiniCssExtractPluginDefault from 'next/dist/build/webpack/plugins/mini-css-extract-plugin';
 
 import type webpack from 'webpack';
@@ -60,6 +61,18 @@ const getVanillaExtractCssLoaders = (
       postcss,
       importLoaders: 1,
       modules: false,
+      url: (url: string, resourcePath: string) =>
+        cssFileResolve(
+          url,
+          resourcePath,
+          options.config.experimental?.urlImports,
+        ),
+      import: (url: string, _: any, resourcePath: string) =>
+        cssFileResolve(
+          url,
+          resourcePath,
+          options.config.experimental?.urlImports,
+        ),
     },
   });
 

